### DISCUSSION TITLE
Simulators aren't shown up

### DISCUSSION THREAD

**Discussion by bsalves** [wojciech-kulik/xcodebuild.nvim#171] [Q&A] [answered]
I've been trying to setup a sample project but I facing a empty list when I reach at the "Select device" step.
I followed the step-by-step and I don't know what is going on here. Could anyone help me to solve that?

Here is the step that I'm having the problem:
<img width="762" alt="Screenshot 2024-09-02 at 11 24 35" src="https://github.com/user-attachments/assets/99c33411-e462-43b8-a407-442c1b228d79">

**Comment by bsalves** [comment, marked as answer]
Guys, I managed to make it. It required to select a Command Line Tools Path in the settings Xcode for the first time.

<img width="1582" alt="Screenshot 2024-09-02 at 11 40 52" src="https://github.com/user-attachments/assets/ce8c55ce-266e-4c50-a7fc-0c34c8e79b82">

**Reply by valenyala** [reply]
Can you please explain that step by step? I'm in the same loading devices problem and i don't know what command line tools you are referring to

**Reply by wojciech-kulik** [reply]
Maybe try `xcode-select --install`

**Reply by SamPan1988** [reply]
I try to reinstall the commmand line tools but it still only shows my iPhone and no simulators detected. Could you give me any advice?
![WX20250109-下午45500@2x](https://github.com/user-attachments/assets/298992ee-0536-4330-b268-6011e2b5329a)
![iphone](https://github.com/user-attachments/assets/5fdd30c3-a80d-42f2-8112-436d06f75de3)

and if I use show destination, I only got the generic options in the list.

![destinations](https://github.com/user-attachments/assets/33159437-fb6a-4d8c-9186-0972c9ef08e2)

**Reply by valenyala** [reply]
@SamPan1988 You can checkout [This issue](https://github.com/nodejs/node-gyp/issues/569) that helped me. (I'm new to the xcode world, i'm not really sure what it's for, but it helps :) )

**Reply by SamPan1988** [reply]
> @SamPan1988 You can checkout [This issue](https://github.com/nodejs/node-gyp/issues/569) that helped me. (I'm new to the xcode world, i'm not really sure what it's for, but it helps :) )

Thank you, but it doesn't fix my problem, it seems my xcode already selects the correct tools.
![image](https://github.com/user-attachments/assets/ecec8dc1-39be-416a-8732-e54e4c69247a)

**Reply by wojciech-kulik** [reply]
The plugin is using the output of xcodebuild command with showdestinations parameter, but I see that even if you call it from cmd you also don’t get results, which suggests that something is wrong with project configuration. First, check if the problem occurs with other projects too. If not see what’s specific about your project configuration. Maybe something is messed up with schemes.

**Reply by wojciech-kulik** [reply]
Also, did you install simulators in Xcode? Does Xcode show you simulators to select for your project?

**Reply by SamPan1988** [reply]
> Also, did you install simulators in Xcode? Does Xcode show you simulators to select for your project?

Yes, I can select simulators in my Xcode, I dont think It's something belong to the project, because it also occur with the one newly created.

**Reply by wojciech-kulik** [reply]
Does `xcode-select -p` points to the right Xcode?

**Reply by SamPan1988** [reply]
> Does `xcode-select -p` points to the right Xcode?

![image](https://github.com/user-attachments/assets/ef5e4b9f-dc21-4957-afa7-7e0d07eeb2b9)
It looks corrected? And I only got one Xcode

**Reply by wojciech-kulik** [reply]
Have you ever built that app via Xcode?

**Reply by wojciech-kulik** [reply]
- I downloaded the project https://github.com/pd95/SwiftUI-Landmarks
- I run `xcodebuild -project Landmarks.xcodeproj -scheme Landmarks -showdestinations`
- I got only generic devices
- I opened Xcode and tried to build app for iOS simulator, it asked me to download WatchOS SDK and I did
- I built the app
- I run again `xcodebuild -project Landmarks.xcodeproj -scheme Landmarks -showdestinations`
- Simulators appeared in the output

If after that it still doesn't work, I can't help. Please report the issue to Apple using Feedback Assistant, because this is an Xcode-related issue if the `xcodebuild` command doesn't return the expected results.

**Reply by SamPan1988** [reply]
> * I downloaded the project https://github.com/pd95/SwiftUI-Landmarks
> * I run `xcodebuild -project Landmarks.xcodeproj -scheme Landmarks -showdestinations`
> * I got only generic devices
> * I opened Xcode and tried to build app for iOS simulator, it asked me to download WatchOS SDK and I did
> * I built the app
> * I run again `xcodebuild -project Landmarks.xcodeproj -scheme Landmarks -showdestinations`
> * Simulators appeared in the output
> 
> If after that it still doesn't work, I can't help. Please report the issue to Apple using Feedback Assistant, because this is an Xcode-related issue if the `xcodebuild` command doesn't return the expected results.

Thank you, I try to replicate your process, but still got only generic output, I guess there's something wrong in my Xcode or system setup.

**Reply by wojciech-kulik** [reply]
I accidentally found some crazy thread on Apple Forum describing the same issue. The workaround was to add ANY Swift Package as a dependency XD. I wonder if it is going to fix it on your side

**Reply by wojciech-kulik** [reply]
Here is the thread: https://developer.apple.com/forums/thread/771110

**Reply by SamPan1988** [reply]
> I accidentally found some crazy thread on Apple Forum describing the same issue. The workaround was to add ANY Swift Package as a dependency XD. I wonder if it is going to fix it on your side

I try but still no luck, though the simulators would be showed in one of my really old project, maybe it's related to some kind of project setting

**Reply by Furthermoore** [reply]
I'm experiencing the same exact thing as @SamPan1988 ... any new xcode projects show empty device lists while older projects show all the simulators.

**Reply by Furthermoore** [reply]
Okay, `rm -rf .nvim` in my project root and now I've got the full list!  Add at least swift package dependency and then xcodebuild.nvim just needs a bit of cleaning

**Reply by wojciech-kulik** [reply]
Weird, maybe an empty list of devices has been cached. In that case, there is a shortcut `<C-r>` to reload list of devices.

**Reply by SamPan1988** [reply]
> > I accidentally found some crazy thread on Apple Forum describing the same issue. The workaround was to add ANY Swift Package as a dependency XD. I wonder if it is going to fix it on your side
> 
> I try but still no luck, though the simulators would be showed in one of my really old project, maybe it's related to some kind of project setting



> I accidentally found some crazy thread on Apple Forum describing the same issue. The workaround was to add ANY Swift Package as a dependency XD. I wonder if it is going to fix it on your side

After struggled to add the swift package properly, I can confirm this workaround would fix the simulator missing problem, thank you.

**Reply by robertherdzik** [reply]
I had the same issue, and according to the suggestion [from here](https://github.com/wojciech-kulik/xcodebuild.nvim/discussions/171#discussioncomment-11923395) i just added random dependency and it works... 🤯
